src/trio/_core/_instrumentation.py

```diff
@@ -98,7 +98,7 @@ def call(self, hookname: str, *args: Any) -> None:
         for instrument in list(self[hookname]):
             try:
                 getattr(instrument, hookname)(*args)
-            except BaseException:
+            except BaseException:  # noqa: PERF203  # try-except in loop
```

Comment:
I'm surprised ruff doesn't like this. Like yeah, if we have a bunch of things to do and some might fail, then the try except belongs *in* the loop.

Reply:
It's a rule we might want to turn off, it just warns whenever a try-except is found inside a loop. The recommendation is to try to refactor so that it's moved around the loop, but that's not always possible. It's also not applicable in 3.11+ due to "zero cost" exception handling.

Reply:
https://docs.astral.sh/ruff/rules/try-except-in-loop/ does say
> This rule is only enforced for Python versions prior to 3.11, which introduced "zero cost" exception handling.

But I suppose we have ruff set to minimum supported python version. In either case the rule documentation also states
> Note that, as with all perflint rules, this is only intended as a micro-optimization, and will have a negligible impact on performance in most cases.

so we can be quite liberal with disabling rules.

Reply:
Disabled `PERF203` in e38b663